newIDE/app/src/VariablesList/useRefocusField.js

```diff
@@ -19,16 +19,19 @@ const useRefocusField = (fieldRefs: {|
     fieldToFocus.current = options;
   }, []);
 
-  React.useLayoutEffect(() => {
-    if (fieldToFocus.current) {
-      const fieldRef = fieldRefs.current[fieldToFocus.current.identifier];
-      if (fieldRef) {
-        fieldRef.focus({ caretPosition: fieldToFocus.current.caretPosition });
+  React.useLayoutEffect(
+    () => {
+      if (fieldToFocus.current) {
+        const fieldRef = fieldRefs.current[fieldToFocus.current.identifier];
+        if (fieldRef) {
+          fieldRef.focus({ caretPosition: fieldToFocus.current.caretPosition });
+        }
       }
-    }
 
-    fieldToFocus.current = null;
-  });
+      fieldToFocus.current = null;
+    },
+    [fieldRefs]
```

Comment:
Not sure it's a good practice, there is a different meaning. Before it was "please run this effect after every DOM update". Now you're telling "please run this effect after a DOM update if fieldRefs changed", which is not the case.

So I think here we're creating a bug/breaking the re-focusing?

Reply:
I agree with 4ian, I think that refs should be in dependencies only for very particular cases. This one is not a particular one: we don't want to call the effect if refs changes, otherwise we wouldn't have used a ref

Reply:
yep, I'll remove this